spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java

```diff
@@ -45,12 +51,30 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 				.getConnection(this.redisConnectionFactory);
 		try {
 			Properties info = connection.info();
-			builder.up().withDetail("version", info.getProperty("redis_version"));
+			if (connection instanceof RedisClusterConnection) {
+				redisClusterInfo(builder,
+						((RedisClusterConnection) connection).clusterGetNodes(), info);
+			}
+			else {
+				defaultRedisInfo(builder, VERSION, info.getProperty(REDIS_VERSION));
+			}
 		}
 		finally {
 			RedisConnectionUtils.releaseConnection(connection,
 					this.redisConnectionFactory);
 		}
 	}
 
+	private void defaultRedisInfo(Health.Builder builder, String key, String value) {
+		builder.up().withDetail(key, value);
+	}
+
+	private void redisClusterInfo(Health.Builder builder,
+			Iterable<RedisClusterNode> redisNodes, Properties info) {
+		for (RedisNode redisNode : redisNodes) {
+			defaultRedisInfo(builder, redisNode.asString() + "." + VERSION,
+					info.getProperty(redisNode.asString() + "." + REDIS_VERSION));
```

Comment:
`info.getProperty(…)` can return `null` when using `redisNode.asString()` because of the execution of the `INFO` command. `INFO` is executed only on master nodes, not on slaves. `clusterGetNodes()` returns all nodes. Passing `null` into `withDetail(key, value)` leads to an `IllegalArgumentException`.

I'd not put too much knowledge about cluster execution in here but rather iterate over the `info` keys (e.g. filter on suffix `"." + VERSION`) and use the keys from `info` to retrieve data.

I'd also propose adding nodes as list to the Redis info as topology details are valuable in a health indicator.